### PR TITLE
[FIX] account: accrued entry account with fiscal position

### PR DESCRIPTION
**Pull request by guva-odoo**
We take the accounts from fiscal position into account
when open accrued entry wizard.

Steps to reproduce:

- Create a product with income (INC) and expense (EXP) account set
- Create a fiscal position which map accounts like this:
  INC -> INC_2
  EXP -> EXP_2
- Create a partner with the fiscal position above.
- Create a purchase/sale order for the product, confirm and
  receive.
- On action menu, select "Accrued Expense/Revenue Entry"
-> The account for the product line on the wizard is not the one from
   fiscal position, it should be.

With this commit we take benefits of get_product_accounts method,
which take fiscal position into account.

opw-2926380

### CODE REVIEW

File: addons/account/wizard/accrued_orders.py [hunk 1]
```diff
@@ -136,10 +143,7 @@ def _ellipsis(string, size):
             if len(orders) == 1 and self.amount:
                 total_balance = self.amount
                 order_line = order.order_line[0]
```
Comment (william-andre) [inline addons/account/wizard/accrued_orders.py:145]: Wtf is this?
So if there are multiple orders, we consider all the lines but not when there is only one?
And we don't care if the first line is a display line or not?

Let's fix this in another commit...

Comment (guva-odoo) [reply addons/account/wizard/accrued_orders.py]: Hi @william-andre

Display_type check is done with last commit.
However I'm not sure that we want to change the behaviour of this condition `if len(orders) == 1 and self.amount`.
As I understand the flow, we pass in this condition when having nothing to invoice (unit_price == 0 on each order line) and set the amount manually. As we need product_id to get the receivable/expense accounts, we have to chose arbitrary one line. Or we can decide to set default accounts for this case if you prefer

File: addons/account/wizard/accrued_orders.py [hunk 2]
```diff
@@ -135,11 +142,8 @@ def _ellipsis(string, size):
         for order in orders:
             if len(orders) == 1 and self.amount:
                 total_balance = self.amount
-                order_line = order.order_line[0]
-                if is_purchase:
-                    account = order_line.product_id.property_account_expense_id or order_line.product_id.categ_id.property_account_expense_categ_id
-                else:
-                    account = order_line.product_id.property_account_income_id or order_line.product_id.categ_id.property_account_income_categ_id
+                order_line = order.order_line.filtered(lambda l: not l.display_type)[0]
```
Comment (william-andre) [inline addons/account/wizard/accrued_orders.py:145]: But why the `[0]`? And why a special case when there is only one order?

Comment (guva-odoo) [reply addons/account/wizard/accrued_orders.py:145]: the [0] is for the reason I explained above. Now, for the special case of len(orders) == 1:

Let's assume an order with 0 quantity received, if we select only this order, we are able to select an amount in the wizard and then we pass in the condition to select the account linked to the first order line.
If we select orders in a batch, we're not allow to chose any amount, therefore they will be no line on the wizard for the order with 0 quantity received.
If we delete this check and do only the instructions in else statement, there will be no line on the wizard when setting the amount manually.

File: addons/account/wizard/accrued_orders.py (hunk 1 above)
Comment (william-andre) [reply addons/account/wizard/accrued_orders.py]: > As I understand the flow, we pass in this condition when having nothing to invoice (unit_price == 0 on each order line)

I don't see where

Comment (guva-odoo) [reply addons/account/wizard/accrued_orders.py]: I meant, when quantity == 0
image

**Comment by guva-odoo** [comment]
Hello @william-andre 

Here is the point about this one

The first commit is about setting the account of the accrued entry regarding the fiscal position of the partner, which is the original object of this PR.

The second is about the first condition `len(orders) == 1` you asked me to fix.

We enter in this condition only in one case : when trying to create an entry from one order with either 0 qty or 0 qty received on each lines.
In this case, if we have only one order, we display the field `amount` in order to create the counterpart line of the manual entry.
IMO, we have several solutions:
- We can delete the condition `len(orders) == 1` but this will not change anything as we not display the field amount because of the _compute_display_amount, therefore we can not create the move because there will be no lines.
- We can change the _compute_display_amount in order to display the field even is there is >1 empty orders, but in this case we have to break at the end of the if statement, otherwise we will have an unbalanced move with
  as many counterpart lines as orders.
- Or we can just let this line as it is, it comes to the same as the first solution.

What do you think about this?

**Comment by william-andre** [comment]
The amount inputted manually should be used with the prorata of the amount on the lines.
If there are multiple accounts that can be selected, it shouldn't be depending on the order of the lines.

**Comment by guva-odoo** [comment]
If we can input the amount manually, that means there is no amount on the lines.
Just to be sure we speak about the same thing,  that the account we can select manually is the one for the counterpart line, not the manual entry. Of what it should depend if not the order of the lines?

**Comment by william-andre** [comment]
> If we can input the amount manually, that means there is no amount on the lines.

No. It means that there is only one order
https://github.com/odoo/odoo/blob/d9872ecce78d6d7ad02cdaa16782aa473e5a7655/addons/account/wizard/accrued_orders.py#L59-L65

**Comment by guva-odoo** [comment]
> No. It means that there is only one order

And no lines

**Comment by william-andre** [comment]
> And no lines

Can you link me to which part of the code ensures that?

**Comment by guva-odoo** [comment]
> > And no lines
> 
> Can you link me to which part of the code ensures that?

https://github.com/odoo/odoo/blob/d9872ecce78d6d7ad02cdaa16782aa473e5a7655/addons/account/wizard/accrued_orders.py#L65

**Comment by guva-odoo** [comment]
fix no need anymore https://github.com/odoo/odoo/pull/96492

**Pull request closed without merging**